File: back/src/models/Notification.js
import mongoose from 'mongoose';

const notificationSchema = new mongoose.Schema(
  {
    id: { type: String, index: true },
    type: { type: String, default: 'general' },
    message: { type: String, required: true },
    targetUser: { type: String, default: 'all', index: true },
    sourceUser: { type: String, default: 'system' },
    read: { type: Boolean, default: false },
    timestamp: { type: String, default: () => new Date().toISOString() },
    serviceYear: { type: String, default: '2026', index: true },
  },
  {
    timestamps: true,
    strict: false,
    collection: 'notifications',
  }
);

notificationSchema.index({ targetUser: 1, read: 1 });

const Notification = mongoose.models.Notification || mongoose.model('Notification', notificationSchema);
export default Notification;

File: back/Models/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("notifications")]
    public class Notification
    {
        [Key]
        [Column("id")]
        public string Id { get; set; } = string.Empty;

        [Column("type")]
        public string Type { get; set; } = "general";

        [Column("message")]
        public string Message { get; set; } = string.Empty;

        [Column("targetUser")]
        public string TargetUser { get; set; } = "all";

        [Column("sourceUser")]
        public string SourceUser { get; set; } = "system";

        [Column("read")]
        public bool Read { get; set; } = false;

        [Column("timestamp")]
        public string Timestamp { get; set; } = DateTime.UtcNow.ToString("o");

        [Column("serviceYear")]
        public string ServiceYear { get; set; } = "2026";

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
